code/modules/telesci/gps.dm

```diff
@@ -179,7 +188,8 @@ var/list/SPS_list = list()
 
 	for(var/E in SPS_list)
 		var/obj/item/device/gps/secure/S = E //No idea why casting it like this makes it work better instead of just defining it in the for each
-		S.announce(wearer, src, "has detected the death of their wearer",dead=TRUE)
+		if (transmitting == TRUE)
```

Comment:
just a style suggestion: `if(transmitting)` does the exact same thing, also:
why don't you move this check near the beginning of the proc? The value of `transmitting` isn't going to change here.
that means turning `if(emped)` into `if(emped || !transmitting)`.
At this point the `emped` check seems superfluous, no? If `emped == TRUE`, `transmitting` will also always be `FALSE`, no?
This needs some more thought.

Reply:
are you saying to just check for `transmitting` and not even bother with emped?

Reply:
yes.